test/src/injector.spec.js

```diff
@@ -175,18 +186,126 @@ describe('Injector', () => {
+            injector.injectIntoDOM([tags]);
+            expect(domUtils.findCssByDataUrl(urls.printed)).to.have.length.above(0);
+          });
+        });
+
+        it('should inject fallback to non-printed css on error', () => {
+          let tags = css.tags(urls);
+
+          return tags.then(tags => {
+            injector.injectIntoDOM([tags]);
+            let errorEvent = new Event('error');
+            domUtils.findCssByDataUrl(urls.printed)[0].dispatchEvent(errorEvent);
+
+            expect(domUtils.findCssByDataUrl(urls.raw)).to.have.length.above(0);
+
+          });
+        });
+
+        it('should inject cached version of asset if provided', () => {
+          cache.set(code, 'css', urls.printed);
+
+          let tags = css.tags(urls);
+
+          return tags.then(tags => {
+            injector.injectIntoDOM([tags]);
+            expect(domUtils.findCssByDataUrl(urls.printed)).to.have.length.above(0);
+            expect(domUtils.findCssByDataUrl(urls.printed)[0].innerHTML).to.equal(code);
+          });
+        });
+      });
+
+      describe('JS', () => {
+        let urls, code;
+
+        before(() => {
+          code = 'var a = "b"';
+          urls = {
+            printed: 'hashed-js-inject.js',
+            raw: 'raw-js-inject.js'
+          }
+        });
+
+        it('should inject fingerprinted JS into the DOM', () => {
+          let tags = js.tags(urls);
+
+          return tags.then(tags => {
+            injector.injectIntoDOM([tags]);
+            expect(domUtils.findJsByDataUrl(urls.printed)).to.have.length.above(0);
+          });
+        });
+
+        it('should inject fallback to non-printed JS on error', () => {
+          let tags = js.tags(urls);
+
+          return tags.then(tags => {
+            injector.injectIntoDOM([tags]);
+            let errorEvent = new Event('error');
+            domUtils.findJsByDataUrl(urls.printed)[0].dispatchEvent(errorEvent);
+
+            expect(domUtils.findJsByDataUrl(urls.raw)).to.have.length.above(0);
+
+          });
+        });
+
+        it('should inject cached version of asset if provided', () => {
+          cache.set(code, 'js', urls.printed);
+
+          let tags = js.tags(urls);
+
+          return tags.then(tags => {
+            injector.injectIntoDOM([tags]);
+            expect(domUtils.findJsByDataUrl(urls.printed)).to.have.length.above(0);
+            expect(domUtils.findJsByDataUrl(urls.printed)[0].innerHTML).to.contain(code);
+          });
+        });
+      });
+
+      describe('multiple assets', () => {
+        let cssUrls, jsUrls;
+
+        before(() => {
+          cssUrls = {
+            printed: 'hashed-css-inject.css',
+            raw: 'raw-css-inject.css'
+          };
+          jsUrls = {
+            printed: 'hashed-js-inject.js',
+            raw: 'raw-js-inject.js'
+          };
+        });
+
+        xit('should inject multiple assets some existing some not as expected', () => {
```

Comment:
ahem yes ... and tests fail because IE once again is not in the mood to play ... I'll take care of it